### ISSUE TITLE
test failure when RoCM device is not available

### ISSUE THREAD

**Issue by yosefe** [openucx/ucx#1683]
When compiling UCX with RoCM libs and running gtest on a machine w/o GPU, following error happens:
```
[1499432526.387011] [vegas10:20348:0]    rocm_common.c:184  UCX  ERROR Failure to open HSA connection: 0x1008
[1499432526.387038] [vegas10:20348:0]    rocm_cma_md.c:58   UCX  ERROR Could not initialize ROCm support
...
<running tests>
...
[----------] 6 tests from rocm/test_md
[ RUN      ] rocm/test_md.reg_advise/0
[     SKIP ] (registration&advise is not supported by rocm)
[       OK ] rocm/test_md.reg_advise/0 (0 ms)
[ RUN      ] rocm/test_md.alloc/0
[     SKIP ] (allocation is not supported by rocm)
[       OK ] rocm/test_md.alloc/0 (0 ms)
[ RUN      ] rocm/test_md.reg/0
[vegas10:20348:0] Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))
[vegas10:20348:0] Process frozen, press Enter to attach a debugger...


```

**Comment by yosefe** [comment]
@ssagalov pls take a look

**Comment by ssagalov** [comment]
I am looking at this right now.

**Comment by ssagalov** [comment]
I found the issue:  There is call to register md (md_ops->mem_reg)  and we crash inside ROCm run-time library. I will add validation in uct_rocm_cma_md_open() to check that rocm is initialized but I have question: uct_rocm_cma_query_md_resources() fails returning UCS_ERR_NO_DEVICE. I  assumed that it means that "mem_reg" should not be called in this case. Did I miss anything?

**Comment by yosefe** [comment]
Need to return UCS_OK, but number of devices should be set to 0

**Comment by ssagalov** [comment]
I changed "mem_reg" according your comments (return UCS_OK and number devices == 0) but "md_reg" is still called. I will add checking in md_reg and send poll request for comments.

**Comment by yosefe** [comment]
@ssagalov does uct_rocm_cma_query_md_resources() set *num_resources_p=0 now?

**Comment by ssagalov** [comment]
Yes. I set *num_resources_p=0  and *resources_p = NULL and return UCS_OK

**Comment by ssagalov** [comment]
Unfortunately it looks like that the issue is a little more complicated than I thought. I need to talk with libraries team which will take some time.

**Comment by yosefe** [comment]
fixes by #1684
